File: app-android/app/src/main/java/com/health2u/data/mapper/UserMapper.kt
package com.health2u.data.mapper

import com.health2u.data.local.entity.UserEntity
import com.health2u.data.remote.dto.UserProfileDto
import com.health2u.domain.model.User

// Entity to Domain
fun UserEntity.toDomain(): User = User(
    id = id,
    email = email,
    name = name,
    profilePictureUrl = profilePictureUrl,
    dateOfBirth = dateOfBirth,
    phone = phone
)

// Domain to Entity
fun User.toEntity(lastSyncTimestamp: Long = System.currentTimeMillis()): UserEntity = UserEntity(
    id = id,
    email = email,
    name = name,
    profilePictureUrl = profilePictureUrl,
    dateOfBirth = dateOfBirth,
    phone = phone,
    lastSyncTimestamp = lastSyncTimestamp
)

// DTO to Domain
fun UserProfileDto.toDomain(): User = User(
    id = id,
    email = email,
    name = name,
    profilePictureUrl = profilePictureUrl,
    dateOfBirth = dateOfBirth,
    phone = phone
)

// Domain to DTO
fun User.toDto(): UserProfileDto = UserProfileDto(
    id = id,
    email = email,
    name = name,
    profilePictureUrl = profilePictureUrl,
    dateOfBirth = dateOfBirth,
    phone = phone
)

File: app-ios/Sources/Health2u/Data/Mapper/UserMapper.swift
import Foundation

extension UserDTO {
    public func toDomain() -> User {
        User(
            id: id,
            email: email,
            name: name,
            profilePictureUrl: profilePictureUrl,
            dateOfBirth: dateOfBirth,
            phone: phone,
            gender: gender,
            heightCm: heightCm,
            weightKg: weightKg,
            bloodType: bloodType,
            hasDiabetes: hasDiabetes,
            hasAllergies: hasAllergies
        )
    }
}

extension User {
    public func toDTO() -> UserDTO {
        UserDTO(
            id: id,
            email: email,
            name: name,
            profilePictureUrl: profilePictureUrl,
            dateOfBirth: dateOfBirth,
            phone: phone,
            gender: gender,
            heightCm: heightCm,
            weightKg: weightKg,
            bloodType: bloodType,
            hasDiabetes: hasDiabetes,
            hasAllergies: hasAllergies
        )
    }

    public func toEntity() -> UserEntity {
        UserEntity(
            id: id,
            email: email,
            name: name,
            profilePictureUrl: profilePictureUrl,
            dateOfBirth: dateOfBirth,
            phone: phone,
            gender: gender,
            heightCm: heightCm,
            weightKg: weightKg,
            bloodType: bloodType,
            hasDiabetes: hasDiabetes,
            hasAllergies: hasAllergies
        )
    }
}

extension UserEntity {
    public func toDomain() -> User {
        User(
            id: id,
            email: email,
            name: name,
            profilePictureUrl: profilePictureUrl,
            dateOfBirth: dateOfBirth,
            phone: phone,
            gender: gender,
            heightCm: heightCm,
            weightKg: weightKg,
            bloodType: bloodType,
            hasDiabetes: hasDiabetes,
            hasAllergies: hasAllergies
        )
    }
}
